include/seqan3/range/view/persist.hpp

```diff
@@ -0,0 +1,318 @@
+
+#pragma once
+
+#include <range/v3/algorithm/copy.hpp>
+
+#include <seqan3/core/metafunction/iterator.hpp>
+#include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/io/exception.hpp>
+#include <seqan3/range/concept.hpp>
+#include <seqan3/range/container/concept.hpp>
+#include <seqan3/range/view/detail.hpp>
+#include <seqan3/range/detail/inherited_iterator_base.hpp>
+#include <seqan3/std/concepts>
+#include <seqan3/std/view/view_all.hpp>
+
+namespace seqan3::detail
+{
+
+// ============================================================================
+//  view_persist
+// ============================================================================
+
+/*!\brief The type returned by seqan3::view::persist.
+ * \tparam urng_t    The type of the underlying ranges, must satisfy std::ranges::InputRange.
+ * \implements std::ranges::View
+ * \implements std::ranges::RandomAccessRange
+ * \implements std::ranges::SizedRange
+ * \ingroup view
+ *
+ * \details
+ *
+ * Note that most members of this class are generated by ranges::view_interface which is not yet documented here.
+ */
+template <std::ranges::InputRange urng_t>
+class view_persist : public ranges::view_interface<view_persist<urng_t>>
+{
+private:
+    //!\brief Shared storage of the underlying range.
+    std::shared_ptr<urng_t> urange;
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief The reference_type.
+    using reference         = reference_t<urng_t>;
+    //!\brief The const_reference type is equal to the reference type if the underlying range is const-iterable.
+    using const_reference   = std::conditional_t<const_iterable_concept<urng_t>, reference, void>;
+    //!\brief The value_type (which equals the reference_type with any references removed).
+    using value_type        = value_type_t<urng_t>;
+    //!\brief The size_type is `size_t` if the the view is exact, otherwise void.
+    using size_type         = std::conditional_t<std::ranges::SizedRange<urng_t>, size_type_t<urng_t>, void>;
+    //!\brief A signed integer type, usually std::ptrdiff_t.
+    using difference_type   = difference_type_t<urng_t>;
+    //!\brief The iterator type of this view (a random access iterator).
+    using iterator          = iterator_t<urng_t>;
+    //!\brief The const_iterator type is equal to the iterator type if the underlying range is const-iterable.
+    using const_iterator    = std::conditional_t<const_iterable_concept<urng_t>, iterator, void>;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    view_persist() = default;
+    constexpr view_persist(view_persist const & rhs) = default;
+    constexpr view_persist(view_persist && rhs) = default;
+    constexpr view_persist & operator=(view_persist const & rhs) = default;
+    constexpr view_persist & operator=(view_persist && rhs) = default;
+    ~view_persist() = default;
+
+    /*!\brief Construct from another range.
+     * \param[in] _urange The underlying range.
```

Comment:
tparam missing

Reply:
it doesnt have a template parameter so I don't need to document it :laughing: